### PR TITLE
Ferigis.129082857.GitHub integration

### PR DESCRIPTION
**Pull request by ferigis**

### CODE REVIEW

File: src/handler/spellingci_callback_handler.erl [hunk 1]
```diff
@@ -90,3 +98,19 @@ access_token(Code) ->
     {error, Reason} ->
       {error, Reason}
   end.
+
+-spec save_user(binary()) -> spellingci_users:token() | undefined.
+save_user(Token) ->
```
Comment (elbrujohalcon) [inline src/handler/spellingci_callback_handler.erl]: Can we put all this logic into `spellingci_users_repo` ?

File: src/model/spellingci_users_repo.erl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec add_days(calendar:datetime(), integer()) -> calendar:datetime().
+add_days({Date, Time}, Days) ->
+  DaysToDate = calendar:date_to_gregorian_days(Date) + Days,
+  NewDate = calendar:gregorian_days_to_date(DaysToDate),
+  {NewDate, Time}.
+
+-spec uuid() -> binary().
+uuid() ->
```
Comment (elbrujohalcon) [inline src/model/spellingci_users_repo.erl]: Why not using [uuid](https://hex.pm/packages/uuid_erl), instead or writing your own uuid function here?

File: test/spellingci_SUITE.erl [hunk 3]
```diff
@@ -105,7 +141,14 @@ test_connection() ->
   true = (binary_to_integer(ContentLength) > 0),
   ok.
 
+-spec call(iodata()) -> {ok, integer(), list(), term()}
+                      | {ok, integer(), list()}
+                      | {error, term()}.
 call(Url) ->
```
Comment (harenson) [inline test/spellingci_SUITE.erl]: I think `api_call` would be a more descriptive name for this function.

File: src/model/spellingci_users_repo.erl [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  Now = calendar:universal_time(),
+  Expires = add_days(Now, 1),
+  UUID = uuid(),
+  User1 = spellingci_users:auth_token(User, UUID),
+  spellingci_users:auth_expires(User1, Expires).
+
+-spec uuid() -> binary().
+uuid() ->
+  State = uuid:new(self()),
+  {Uuid, _} = uuid:get_v1(State),
```
Comment (harenson) [inline src/model/spellingci_users_repo.erl]: I thought we were going to use `v5`.

**Pull request merged**